### PR TITLE
Add representation language to address syntax and "unknown" properties.

### PR DESCRIPTION
**Pull request by dlongley**
Here's an attempt at getting generic/unknown/syntax-related property representation rules into the spec per the resolutions/discussions this past week.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/pull/454.html" title="Last updated on Dec 1, 2020, 6:25 PM UTC (35e6c03)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/454/b86bed4...35e6c03.html" title="Last updated on Dec 1, 2020, 6:25 PM UTC (35e6c03)">Diff</a>

### CODE REVIEW

**Comment by dlongley** [comment]
This PR may need some further tweaks/rebasing on another PR that separates properties in the data model that carry representation-syntax from information properties (with information related to the DID subject). I'll coordinate with @msporny on that restructuring as needed.

**Comment by dlongley** [comment]
See #455 for reworked text around property types. This can be rebased on top of that PR when appropriate.

**Review by kdenhartog — APPROVED** [review]

File: index.html
Comment (OR13) [inline index.html:1166]: do we need a definition for "representation-specific syntax" ?

**Review by OR13 — APPROVED** [review]

File: index.html
Comment (brentzundel) [inline index.html:1168]:
```suggestion
included when consuming or producing to aid in lossless conversion.
```

File: index.html
Comment (brentzundel) [inline index.html:2459]: I think this is to address the possible `"@context": true` scenario. If so, it may be clearer to say: 
```suggestion
explicitly provided to the producer or if the value from the abstract data
model is not valid according to the JSON-LD Representation's consumption rules.
```

**Review by brentzundel — APPROVED** [review]
Some editorial suggestions, otherwise LGTM.

File: index.html
Comment (dlongley) [reply index.html:1166]: This may get editorially adjusted to "representation properties" (or whatever language is settled on there) based on PR #455.

**Review by talltree — APPROVED** [review]

**Review by peacekeeper — APPROVED** [review]
See my comment here https://github.com/w3c/did-core/pull/455#pullrequestreview-529511324. I agree this reflects the decisions during the F2F. Still, for the record, it doesn't feel great to call representation-specific syntax "properties" in the same way as verification methods, services, etc. are "properties".

**Review by rhiaro — APPROVED** [review]

File: index.html
Comment (jonnycrunch) [inline index.html:2252]: The Abstract Data Model is too Abstract. We need to make this more concrete before enforcing the MUST.

File: index.html
Comment (selfissued) [inline index.html:2383]: Please replace the deleted text above with the text of the resolution https://www.w3.org/2019/did-wg/Meetings/Minutes/2020-11-03-did#resolution2 : "Unrecognized properties MUST be preserved."  If this is not the right place, then put it somewhere else.  But the text of the approved resolution needs to normatively appear in the spec.

File: index.html
Comment (selfissued) [inline index.html:2379]: Please replace the text above with that approved in the resolution https://www.w3.org/2019/did-wg/Meetings/Minutes/2020-11-05-did#resolution2 : "Note that the value of the @context object member will be ignored, if present."

**Review by selfissued — CHANGES_REQUESTED** [review]
I cited text from two resolutions cited that needs to appear in the PR, for it to do the jobs that the working group agreed to.  Thank you.

File: index.html
Comment (jonnycrunch) [reply index.html:1166]: is CDDL "representation-specific syntax"?

Comment (OR13) [reply index.html:1166]: @jonnycrunch based on the relationship between CDDL and mime types, I would say that CDDL is an expression of representation specific syntax, that is more compatible with a comprehensible ADM.

File: index.html
Comment (brentzundel) [inline index.html:2344]: @selfissued @msporny would this be acceptable?
```suggestion
        </p>
        <p class="note" title="@context">
"Please note that the value of the @context object member will be ignored, if present."
        </p>

```

File: index.html
Comment (brentzundel) [inline index.html:2396]: @selfissued @msporny would this be acceptable?
```suggestion
        </p>
        <p class="note" title="@context">
"Please note that the value of the @context object member will be ignored, if present."
        </p>

```

Comment (jricher) [inline index.html:2396]:
```suggestion
        <p>
Note that the value of the <code>@context</code> object member will be ignored, if present. That is, this field will not have additional processing applied to its value and will be added verbatim to the abstract data model.
        </p>
```

File: index.html
Comment (jricher) [inline index.html:2250]:
```suggestion
Unrecognized properties MUST be preserved. An unrecognized property is
any property that does not have explicit processing rules known to the consumer or producer.
Consumers MUST add all properties that do not have explicit processing rules
```

**Review by jricher — CHANGES_REQUESTED** [review]

File: index.html
Comment (brentzundel) [reply index.html:2252]: I think we did that with #455

File: index.html
Comment (brentzundel) [reply index.html:2383]: I think the spirit of that resolution may be captured by this text, which is proposed above:
```
Consumers MUST add all properties that do not have explicit processing rules
for the representation being consumed to the abstract data model using only the
representation's generic type processing rules. Producers MUST serialize all
properties in the abstract data model that do not have explicit processing
rules for the representation being produced using only the representation's
generic type processing rules.
```

File: index.html
Comment (dlongley) [reply index.html:2396]: It reintroduces the problematic language. Here's another option that could go into this PR and then it can be further tweaked by editors after merge:

```suggestion
        </p>
        <p class="note" title="@context">
"Please note that the value of the `@context` property is JSON-LD specific representation syntax and must be treated according to the generic representation production and consumption rules."
        </p>
```

**Comment by burnburn** [comment]
Apologies all, I misread [resolution #2](https://www.w3.org/2019/did-wg/Meetings/Minutes/2020-11-05-did#resolution2) during today's call.  We absolutely agreed, in that resolution, to add an *informative* note about the effect on @context, as the text of the resolution specifically says, independent of the normative language we are properly adding elsewhere.  In fact, I was the one to propose that specific text wording be added.  And yes, it is legal to include informative notes in normative sections of specifications (which was the discussion Ivan and I had on the f2f call on the adding of this specific text).

**Comment by dlongley** [comment]
@burnburn, I don't think anyone is opposed to the note, the issue is with the language (we want to avoid reintroducing the ambiguous "perserved"/"ignored" language in the note). I didn't want it to hold up this PR while we figure out exactly what the note language should be or where the note should go at this point. However, since that ship has sailed and it's holding things up now, there's a proposed change with a note in it that can be further editorialized in the future. Hopefully this gets us to consensus and the PR can be merged.

File: index.html
Comment (dlongley) [reply index.html:2396]: I'm also ok with this version of the note. We might want to mention that it follows the same rules as any other representation syntax (it's not special), but that can be done editorially.

File: index.html
Comment (brentzundel) [reply index.html:2250]: @selfissued this suggestion is an attempt to address your concerns, can you provide feedback so we can move this forward?

File: index.html
Comment (brentzundel) [reply index.html:2396]: @selfissued this suggestion is an attempt to address your concerns, can you provide feedback so we can move this forward?

File: index.html
Comment (brentzundel) [reply index.html:2489]: This suggestion may address this concern: https://github.com/w3c/did-core/pull/454/#discussion_r525328136

File: index.html
Comment (brentzundel) [reply index.html:2493]: I believe this suggestion addresses your concerns: https://github.com/w3c/did-core/pull/454/#discussion_r525329052

**Comment by msporny** [comment]
Merge conflicts, please fix.

**Comment by msporny** [comment]
@selfissued can you please respond to the people that are asking you questions in this PR. At this point, there is general agreement to merge it -- you're the one that everyone is waiting on to make progress on this PR.

File: index.html
Comment (TallTed) [reply index.html:2501]: I can't seem to make a suggestion atop the suggestion, so I'll just write it here.
```suggestion
Note that no additional processing will be applied to the value of the 
<code>@context</code> object member, if present. The original value will 
be added verbatim to the abstract data model.
```

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2019/did-wg/Meetings/Minutes/2020-11-24-did#section3) on 2020-11-24)

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 3. PR 454 and 455... 

_See github pull request [#454](https://github.com/w3c/did-core/pull/454), [#455](https://github.com/w3c/did-core/pull/455)._




**Brent Zundel:** status update?  

**dlongely:** we are waiting for a response from mike jones regarding the suggestions for adding a note to the PR / regardign @context in the JSON representation section.  
… unsure if he is still objecting  

**Brent Zundel:** I will ping him again  

**Markus Sabadello:** after merging 455 and then 454 we still need to revisit the exact terminology... lots of different properties type to hammer down.  

_See github issue [#463](https://github.com/w3c/did-core/issues/463)._




**Markus Sabadello:** feedback welcome  

**Brent Zundel:** other PRs that need to be discussed in this meeting?  

</details>

**Review by selfissued — APPROVED** [review]

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merging.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2019/did-wg/Meetings/Minutes/2020-11-01-did#section2) on 2020-11-01)

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 2. PR 454 

_See github pull request [#454](https://github.com/w3c/did-core/pull/454)._




**Daniel Burnett:** main item for today pull request 454  

**Daniel Burnett:** Manu where are we  

**Manu Sporny:** we are waiting for mike jones to tell us if he is happy with it or not  

**Daniel Burnett:** we had some time waiting for proposals. should we move on to priority 1 & 2 issues and come back go 454 (10 min at the end)  

</details>

**Pull request merged**
